metrics/processors/pod_based_enricher.go

```diff
@@ -46,26 +46,22 @@ func (this *PodBasedEnricher) Process(batch *core.DataBatch) (*core.DataBatch, e
 		return nil, err
 	}
 	for _, pod := range pods {
-		addPodInfo(pod, batch)
+		// Only keep stats for pending & running pods. We keep pending pods as well
+		// as running because pending pods might be partially started (e.g. one container
+		// in a multi-container pod).
+		if pod.Status.Phase == kube_api.PodPending || pod.Status.Phase == kube_api.PodRunning {
+			addPodInfo(pod, batch)
+		}
```

Comment:
I wonder what to do if the status is different (deleted/failed/whatever)? Shouldn't we drop the remains?